### PR TITLE
Remove player restriction from entity:setSkin(number)

### PR DESCRIPTION
**Pull request by AshleyJamesy**
Followed similar pattern to how setMaterial and setSubMaterial are setup.

### CODE REVIEW

File: lua/entities/gmod_wire_expression2/core/entity.lua [hunk 1]
```diff
@@ -526,16 +526,13 @@ end
 --- Sets <this>'s skin number.
 e2function void entity:setSkin(skinIndex)
 	if not IsValid(this) then return self:throw("Invalid entity!", nil) end
-	if this:IsPlayer() then return self:throw("You cannot set the skin of a player!", nil) end
+	if not isOwner(self, this) then return self:throw("You do not own this entity!", nil) end
+
+	if this:SkinCount() < 1 then return end
```
Comment (Vurv78) [inline lua/entities/gmod_wire_expression2/core/entity.lua:531]: Might as well add a `self:throw()` here now that you've separated it: would help avoid confusion
```suggestion
	if this:SkinCount() < 1 then return self:throw("This entity does not have any skins", nil) end
```

File: lua/entities/gmod_wire_expression2/core/entity.lua [hunk 2]
```diff
@@ -526,16 +526,13 @@ end
...
+	if skinIndex >= this:SkinCount() then return end
```
Comment (Vurv78) [inline lua/entities/gmod_wire_expression2/core/entity.lua:532]: Same as above
```suggestion
	if skinIndex >= this:SkinCount() then return self:throw("Invalid skin index", nil) end
```

**Review by Vurv78 — CHANGES_REQUESTED** [review]
Idk about removing the `canProperty`, feel like it'll be more restrictive, but if it works for setMaterial then that's fine

File: lua/entities/gmod_wire_expression2/core/entity.lua (hunk 1 above)
Comment (Denneisk) [reply lua/entities/gmod_wire_expression2/core/entity.lua:531]: In my opinion it shouldn't throw if you're out of bounds. It seems like too much.

Comment (AshleyJamesy) [reply lua/entities/gmod_wire_expression2/core/entity.lua:531]: @Vurv78 I agree with @Denneisk - I think an invalid skin index shouldn't do anything, or should reset skin to default. What do you think?

Comment (j7gg) [reply lua/entities/gmod_wire_expression2/core/entity.lua:531]: Yeah @AshleyJamesy agree, just reset skin to default would be better

Comment (AshleyJamesy) [reply lua/entities/gmod_wire_expression2/core/entity.lua:531]: Most of the time the setSkin function is used for player model cosmetics

Comment (Fasteroid) [reply lua/entities/gmod_wire_expression2/core/entity.lua:531]: Throwing an error on `@strict` is a good idea, but I agree with Ashley that passing invalid data should trigger a reset to defaults in the same way functions like `setMaterial` do.

**Comment by AshleyJamesy** [comment]
> I think it's fine either way, just make a resolution to Vurv's suggestions.

What do I need to change to get this merged? Remove the unnecessary index checking as per below?

> In Gmod an OOB skin doesn't check for validity and visually resets to 0 but with its value preserved at the set point. From what I understand, these checks could be removed since it doesn't seem to cause issues.

**Comment by AshleyJamesy** [comment]
@Denneisk Done

**Review by Denneisk — APPROVED** [review]

**Comment by j7gg** [comment]
so is this added yet?

**Comment by Denneisk** [comment]
Don't wanna go alone against Vurv since I'm still new :P

**Comment by Denneisk** [comment]
Will probably merge this at the start of next month if no word from either :P

**Pull request merged**